File: VB/ColorListLegend_SortOrder/Form1.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraMap

Namespace ColorListLegend_SortOrder
    Partial Public Class Form1
        Inherits Form

        Private legend As ColorListLegend

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            legend = New ColorListLegend() With {.Alignment = LegendAlignment.TopRight}
            legend.CustomItems.Add(New ColorLegendItem() With {.Color = Color.Orange, .Text = "First color"})
            legend.CustomItems.Add(New ColorLegendItem() With {.Color = Color.LightGray, .Text = "Second color"})
            legend.CustomItems.Add(New ColorLegendItem() With {.Color = Color.Coral, .Text = "Third color"})
            mapControl.Legends.Add(legend)

            lbSortOrder.DataSource = System.Enum.GetValues(GetType(LegendItemsSortOrder))
        End Sub

        Private Sub lbSortOrder_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles lbSortOrder.SelectedIndexChanged
            legend.SortOrder = DirectCast(lbSortOrder.SelectedValue, LegendItemsSortOrder)
        End Sub
    End Class
End Namespace

File: CS/ColorListLegend_SortOrder/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraMap;

namespace ColorListLegend_SortOrder {
    public partial class Form1 : Form {
        ColorListLegend legend;
        
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            legend = new ColorListLegend() { Alignment = LegendAlignment.TopRight };
            legend.CustomItems.Add(new ColorLegendItem() { 
                Color = Color.Orange, 
                Text = "First color" 
            });
            legend.CustomItems.Add(new ColorLegendItem() {
                Color = Color.LightGray,
                Text = "Second color"
            });
            legend.CustomItems.Add(new ColorLegendItem() {
                Color = Color.Coral,
                Text = "Third color"
            });
            mapControl.Legends.Add(legend);

            lbSortOrder.DataSource = Enum.GetValues(typeof(LegendItemsSortOrder));
        }

        private void lbSortOrder_SelectedIndexChanged(object sender, EventArgs e) {
            legend.SortOrder = (LegendItemsSortOrder)lbSortOrder.SelectedValue;
        }
    }
}
